### PR TITLE
fix: naming and minor structural improvements

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/SecondaryToPrimaryMapper.java [hunk 1]
```diff
@@ -6,5 +6,5 @@
 
 @FunctionalInterface
 public interface SecondaryToPrimaryMapper<T> {
-  Set<ResourceID> associatedPrimaryResources(T dependentResource);
+  Set<ResourceID> toPrimaryResourceID(T dependentResource);
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/SecondaryToPrimaryMapper.java:9]: This should be a plural.

**Review by metacosm — CHANGES_REQUESTED** [review]

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/SecondaryToPrimaryMapper.java (hunk 1 above)
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/SecondaryToPrimaryMapper.java]: fixed, thx

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/controller/ControllerResourceEventSource.java [hunk 2]
```diff
@@ -103,6 +103,6 @@ private void handleKubernetesClientException(Exception e) {
 
   @Override
   public Optional<T> getSecondaryResource(T primary) {
-    return get(ResourceID.fromResource(primary));
+    throw new IllegalStateException("This method should not be called here. Primary: " + primary);
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/controller/ControllerResourceEventSource.java:106]: Why this change?

**Review by metacosm — APPROVED** [review]
LGTM

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/controller/ControllerResourceEventSource.java (hunk 2 above)
Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/controller/ControllerResourceEventSource.java:106]: It does not makes sense imo to ask for the secondary resource from the `ControllerEventSource`. It basically does not have secondary, so this operation should fail.
